android/src/main/kotlin/co/doneservices/callkeep/CallKeep.kt

```diff
@@ -609,4 +611,12 @@ class CallKeep(private val channel: MethodChannel, private var applicationContex
 
         return true
     }
+
+    fun getPendingIntentActivity(context: Context?, id: Int, intent: Intent?, flag: Int): PendingIntent {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            PendingIntent.getActivity(context, id, intent, PendingIntent.FLAG_MUTABLE or flag)
+        } else {
+            PendingIntent.getActivity(context, id, intent, flag)
+        }
```

Comment:
Can't this be written something like below?
```kotlin
PendingIntent.getActivity(context, id, intent, Build.VERSION.SDK_INT >= Build.VERSION_CODES.S ? PendingIntent.FLAG_MUTABLE or flag : flag)
```

Reply:
Btw, why is Google doing this? Introducing a new API and making it required at the same time, so stupid.
https://developer.android.com/reference/android/app/PendingIntent#FLAG_MUTABLE

```
Up until [Build.VERSION_CODES.R](https://developer.android.com/reference/android/os/Build.VERSION_CODES#R), PendingIntents are assumed to be mutable by default, unless [FLAG_IMMUTABLE](https://developer.android.com/reference/android/app/PendingIntent#FLAG_IMMUTABLE) is set. Starting with [Build.VERSION_CODES.S](https://developer.android.com/reference/android/os/Build.VERSION_CODES#S), it will be required to explicitly specify the mutability of PendingIntents on creation with either (@link #FLAG_IMMUTABLE} or [FLAG_MUTABLE](https://developer.android.com/reference/android/app/PendingIntent#FLAG_MUTABLE).
```

Reply:
I have no idea why they decided to discard the default behavior all of a sudden, I'll try the alternative syntax

Reply:
The fuck, that's a stupid compiler :P

Reply:
```kotlin
        return PendingIntent.getActivity(context, id, intent, if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.S ) PendingIntent.FLAG_MUTABLE or flag else flag)
```
This works though